Added new features and small tweaks according to my needs.

motionkartik opened the pull request:
**Hi bgibson72,**

Thank you for creating such a beautiful and functional startpage. I have been using my own lightweight, mobile focused startpage for some time, but for desktop usage I was looking for something more advanced and customizable. Your project fit that need perfectly, and it has been a pleasure working with it. It truly feels like a privilege to build on top of your work.

While adapting the startpage to better suit my personal workflow, I made a few changes and enhancements. I am sharing them here in case you feel any of these could be useful additions to the main branch. Modifying your repository was genuinely fun and insightful.

Summary of changes:

- Repositioned the weather widget
- Added Air Quality Index (AQI) support
- Switching between 12 and 24 hour time formats by clicking on the time
- Removed the help modal (to make it more minimal)
- Removed the temperature unit toggle and kept Celsius only (Celsius is only used in my region)
- Added backup and restore functionality
- Reduced default search engines to two (to make it more minimal)
- Updated regions for some search engines to my country
- Modified the footer to display social links
- Removed keyboard hints (to make it more minimal)
- Removed quotes section (to make it more minimal)
- All links open in a new tab

Thank you again for the excellent project, and for making it open to customization and contribution.

motionkartik committed:
1st Changes

Changed Weather Position
Added AQI
Removed Help Modal
Removed Temperature toggle, kept Celsius only
Added support for Backup / Restore
Reduced default search engines to 2
Changed some search engines regions
Changed footer to show socials
Removed Keyboard hints
Removed Quotes
Links open in new tab
Clicking on time switches between 12 and 24 hours format

motionkartik committed:
1st Changes (Cleaned-up)

Changed Weather Position
Added AQI
Removed Help Modal
Removed Temperature toggle, kept Celsius only
Added support for Backup / Restore
Reduced default search engines to 2
Changed some search engines regions
Changed footer to show socials
Removed Keyboard hints
Removed Quotes
Links open in new tab
Clicking on time switches between 12 and 24 hours format

motionkartik committed:
Fixed a typo while cleaning up

bgibson72 commented:
Hello Kartik!

Thank you for your kind words, and for providing this detailed PR! I would be happy to collab with you on this project. 

Here are some thoughts I had regarding your suggestions:

1. Repositioned the weather widget

I would like to add a setting that allows you to place the widget at the top or bottom of the page.

2. Added Air Quality Index (AQI) support

This is an easy add, and I'd be happy to support adding other variables as well.

3. Switching between 12 and 24 hour time formats by clicking on the time

Great idea!

4. Removed the help modal (to make it more minimal)

Understandable--I was thinking of moving this to a tab on the Settings modal.

5. Removed the temperature unit toggle and kept Celsius only (Celsius is only used in my region)

This would be a personal decision for the enduser, so I'd like to keep the toggle. Savvy users can remove the option on their own as you have done.

6. Added backup and restore functionality

Great feature! I've also though about the ability to maintain additional profiles (work, personal, etc).

7. Reduced default search engines to two (to make it more minimal)

The Settings modal contains a multi-selection of search engines at the bottom of the sheet to easily add or remove these.

8. Updated regions for some search engines to my country

That's a nice feature, and might be expanded by adding various language support.

9. Modified the footer to display social links
10. Removed keyboard hints (to make it more minimal)
11. Removed quotes section (to make it more minimal)

These could all be added as toggles to the Settings modal for added customization.

12. All links open in a new tab

Users who opt to install the Sip extension on Chrome and Firefox will see a new instance of Sip for each tab they open anyway, but for those of us using Sip as a true startpage, this would be a good feature to add!

motionkartik committed:
Changes_v02

Removed categories and links limit
Added ability to pin category
Added Ability to hide categories and links

motionkartik committed:
Search Engine enhancement

Separated the Search engines to its own tab with ability to add custom engines

motionkartik commented:
Absolutely, I never wanted my entire PR to be merged, there were many changes according to my needs only, so merging them with main repo wouldn't have made sense.
there were some ideas which I felt good to be shared, like clock, footer customisation, and most importantly import export feature.
I'm happy you recognised those!

bgibson72 commented:
Of course! I appreciate the collaboration!
